src/equations/3d/compressible_euler.jl

```diff
@@ -330,6 +330,27 @@ function source_terms_eoc_test_coupled_euler_gravity(ut, u, x, element_id, t, n_
   return nothing
 end
 
+@inline function source_terms_eoc_test_coupled_euler_gravity(u, x, t, equation::CompressibleEulerEquations3D)
+  # Same settings as in `initial_condition_eoc_test_coupled_euler_gravity`
+  c = 2.0
+  A = 0.1
+  G = 1.0 # gravitational constant, must match coupling solver
+  C_grav = -4 * G / (3 * pi) # "3" is the number of spatial dimensions  # 2D: -2.0*G/pi
+
+  x1, x2, x3 = x
+  rhox = A * pi * cospi(x1 + x2 + x3 - t)
+  rho  = c + A * sinpi(x1 + x2 + x3 - t)
```

Comment:
While talking about performance... have you tried using `sincos` instead of `cospi` + `sinpi`?

Reply:
Good suggestion! I've done that for `trixi_include("examples/3d/elixir_eulergravity_eoc_test.jl")`.
```julia
  rhox = A * pi * cospi(x1 + x2 + x3 - t)
  rho  = c + A * sinpi(x1 + x2 + x3 - t)
```
results in
```julia
julia> @benchmark Trixi.calc_sources!(
           $(Trixi.wrap_array(copy(sol.u[end]), semi)),
           $(Trixi.wrap_array(copy(sol.u[end]), semi)),
           $(sol.t[end]),
           $(semi.semi_euler.source_terms),
           $(equations_euler),
           $(solver_euler),
           $(semi_euler.cache))
BenchmarkTools.Trial: 
  memory estimate:  928 bytes
  allocs estimate:  6
  --------------
  minimum time:     84.431 μs (0.00% GC)
  median time:      86.245 μs (0.00% GC)
  mean time:        88.844 μs (0.00% GC)
  maximum time:     269.564 μs (0.00% GC)
  --------------
  samples:          10000
  evals/sample:     1
```

```julia
  si, co = sincos(pi * (x1 + x2 + x3 - t))
  rhox = A * pi * co
  rho  = c + A * si
```
results in
```julia
julia> @benchmark Trixi.calc_sources!(
           $(Trixi.wrap_array(copy(sol.u[end]), semi)),
           $(Trixi.wrap_array(copy(sol.u[end]), semi)),
           $(sol.t[end]),
           $(semi.semi_euler.source_terms),
           $(equations_euler),
           $(solver_euler),
           $(semi_euler.cache))
BenchmarkTools.Trial: 
  memory estimate:  928 bytes
  allocs estimate:  6
  --------------
  minimum time:     77.625 μs (0.00% GC)
  median time:      78.515 μs (0.00% GC)
  mean time:        80.168 μs (0.00% GC)
  maximum time:     169.413 μs (0.00% GC)
  --------------
  samples:          10000
  evals/sample:     1
```
Hence, I've changed that in e9746a0 and opened #277 to remind us of using `sincospi` when moving to Julia `v1.6`.

Reply:
Nice! Thanks!